erizo_controller/ROV/rovMetricsGatherer.js

```diff
@@ -0,0 +1,107 @@
+        .then((results) => {
+          let totalRooms = 0;
+          results.forEach((roomsSize) => {
+            totalRooms += parseInt(roomsSize, 10);
+          });
+          this.log.debug(`Total rooms result: ${totalRooms}`);
+          this.prometheusMetrics.activeRooms.set(totalRooms);
+          resolve();
+        })
+        .catch((error) => {
+          this.log.error('Error in getTotalRooms', error);
+          reject(error);
+        });
+    });
+  }
+
+  getTotalClients() {
+    return new Promise((resolve, reject) => {
+      this.log.debug('Getting total clients');
+      this.rovClient.runInComponentList('var totalClients = 0; context.rooms.forEach((room) => {totalClients += room.clients.size}); console.log(totalClients);',
+        this.rovClient.components.erizoControllers)
+        .then((results) => {
+          let totalClients = 0;
+          results.forEach((clientsInRoom) => {
+            totalClients += isNaN(clientsInRoom) ? 0 : parseInt(clientsInRoom, 10);
+          });
+          this.prometheusMetrics.activeClients.set(totalClients);
+          this.log.debug(`Total clients result: ${totalClients}`);
+          resolve();
+        })
+        .catch((error) => {
+          this.log.error('Error in getTotalClients', error);
+          reject(error);
+        });
+    });
+  }
+
+  getTotalPublishersAndSubscribers() {
+    return new Promise((resolve, reject) => {
+      this.log.debug('Getting total publishers and subscribers');
+      const requestPromises = [];
+      const command = 'var totalValues = {publishers: 0, subscribers: 0}; context.rooms.forEach((room)' +
+        '=> {const pubsubList = room.controller.getSubscribers(); const publishersInList = Object.keys(pubsubList); totalValues.publishers += publishersInList.length;' +
+        'publishersInList.forEach((pubId) => {totalValues.subscribers += pubsubList[pubId].length; });}); console.log(JSON.stringify(totalValues));';
+      this.rovClient.components.erizoControllers.forEach((controller) => {
+        requestPromises.push(controller.runAndGetPromise(command));
+      });
+      let totalPublishers = 0;
+      let totalSubscribers = 0;
+      Promise.all(requestPromises).then((results) => {
+        results.forEach((result) => {
+          const parsedResult = JSON.parse(result);
+          totalPublishers += parsedResult.publishers;
+          totalSubscribers += parsedResult.subscribers;
+        });
+        this.prometheusMetrics.totalPublishers.set(totalPublishers);
+        this.prometheusMetrics.totalSubscribers.set(totalSubscribers);
+        this.log.debug(`Total publishers and subscribers result: ${totalPublishers}, ${totalSubscribers}`);
+        resolve();
+      })
+        .catch((error) => {
+          this.log.error('Error in getTotalPublishers', error);
+          reject(error);
+        });
+    });
+  }
+
+  getActiveProcesses() {
```

Comment:
you can just return Promise.resolve() here if you want to give it a Promise-like interface.